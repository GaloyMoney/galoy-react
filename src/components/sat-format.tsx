import { memo } from "react"
import { SatSymbol } from ".."

type Props = {
  amount: number
}

export const satsFormatter = new Intl.NumberFormat("en-US", {
  maximumFractionDigits: 0,
})

export const SatFormatComponent = ({ amount }: Props) => {
  const formattedSats = satsFormatter.format(Math.abs(amount))
  return (
    <span className="sat-format" title="Bitcoin Satoshis">
      {amount < 0 && <span className="minus">-</span>}
      <SatSymbol />
      {formattedSats}
    </span>
  )
}

export const SatFormat = memo(SatFormatComponent)
