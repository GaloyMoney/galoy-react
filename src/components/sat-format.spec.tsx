import { render } from "@testing-library/react"

import { SatFormat } from "./sat-format"

describe("SatFormat", () => {
  it("renders and matches snapshot", () => {
    const { asFragment } = render(<SatFormat amount={420} />)
    expect(asFragment()).toMatchSnapshot()
    // TODO: match 42
  })
})
