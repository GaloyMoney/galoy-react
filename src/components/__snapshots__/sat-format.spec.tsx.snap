// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SatFormat renders and matches snapshot 1`] = `
<DocumentFragment>
  <span
    class="sat-format"
    title="Bitcoin Satoshis"
  >
    <svg
      fill="none"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M18.606 2.86741L17.0688 2L15.5524 4.56685L17.0897 5.43426L18.606 2.86741ZM8.82022 19.4331L7.28298 18.5657L5.76664 21.1326L7.30388 22L8.82022 19.4331ZM20.4706 10.347L19.5831 11.8494L8.62228 5.66464L9.50981 4.16223L20.4706 10.347ZM17.2389 15.8174L18.1265 14.315L7.16567 8.13017L6.27814 9.63257L17.2389 15.8174ZM15.8483 18.1709L14.9608 19.6733L4 13.4885L4.88753 11.9861L15.8483 18.1709Z"
        fill="currentColor"
        fill-rule="evenodd"
      />
    </svg>
    420
  </span>
</DocumentFragment>
`;
